python/lsst/ip/diffim/imageDecorrelation.py

```diff
@@ -52,188 +54,211 @@ def computeClippedImageStats(im):
+        out = np.roll(kernel, kernel.shape[0]//2 - maxloc[0], axis=0)
+        out = np.roll(out, out.shape[1]//2 - maxloc[1], axis=1)
+        # Make sure it is odd-dimensioned by trimming it.
+        if (out.shape[0] % 2) == 0:
+            maxloc = np.unravel_index(np.argmax(out), out.shape)
+            if out.shape[0] - maxloc[0] > maxloc[0]:
+                out = out[:-1, :]
+            else:
+                out = out[1:, :]
+            if out.shape[1] - maxloc[1] > maxloc[1]:
+                out = out[:, :-1]
+            else:
+                out = out[:, 1:]
         return out
 
-    pcf = post_conv_psf(psf=im2_psf, kernel=kappa, sig1=sig2, sig2=sig1)
-    pcf = pcf.real / pcf.real.sum()
-    return pcf
+    def doConvolve(self, exposure, kernel, use_scipy=False):
+        """! Convolve an Exposure with a convolution kernel.
+        @param exposure Input afw.image.Exposure to be convolved.
+        @param kernel Input 2-d numpy.array to convolve the image with
+        @param use_scipy Use scipy to do convolution instead of afwMath
+        @return a new Exposure with the convolved pixels.
+
+        @note we use afwMath.convolve() but keep scipy.convolve for debugging.
+        """
+        outExp = kern = None
+        fkernel = self.fixEvenKernel(kernel)
+        if use_scipy:
+            pci = scipy_convolve(exposure.getMaskedImage().getImage().getArray(),
+                                 fkernel, mode='constant', cval=np.nan)
+            outExp = exposure.clone()
+            outExp.getMaskedImage().getImage().getArray()[:, :] = pci
+            kern = fkernel
 
-
-def fixEvenKernel(kernel):
-    """! Take a kernel with even dimensions and make them odd, centered correctly.
-    @param kernel a numpy.array
-    @return a fixed kernel numpy.array
-    """
-    # Make sure the peak (close to a delta-function) is in the center!
-    maxloc = np.unravel_index(np.argmax(kernel), kernel.shape)
-    out = np.roll(kernel, kernel.shape[0]//2 - maxloc[0], axis=0)
-    out = np.roll(out, out.shape[1]//2 - maxloc[1], axis=1)
-    # Make sure it is odd-dimensioned by trimming it.
-    if (out.shape[0] % 2) == 0:
-        maxloc = np.unravel_index(np.argmax(out), out.shape)
-        if out.shape[0] - maxloc[0] > maxloc[0]:
-            out = out[:-1, :]
         else:
-            out = out[1:, :]
-        if out.shape[1] - maxloc[1] > maxloc[1]:
-            out = out[:, :-1]
-        else:
-            out = out[:, 1:]
-    return out
-
-
-def doConvolve(exposure, kernel, use_scipy=False):
-    """! Convolve an Exposure with a convolution kernel.
-    @param exposure Input afw.image.Exposure to be convolved.
-    @param kernel Input 2-d numpy.array to convolve the image with
-    @param use_scipy Use scipy to do convolution instead of afwMath
-    @return a new Exposure with the convolved pixels.
-
-    @note we use afwMath.convolve() but keep scipy.convolve for debugging.
```

Comment:
I suggest not leaving scipy.convolve for debugging unless you have a really important reason. It clutters up the code and mistreats mask and variance.